src/main/java/com/artipie/asto/cache/Cache.java

```diff
@@ -39,7 +40,17 @@
     /**
      * No cache, just load remote resource.
      */
-    Cache NOP = (key, remote, ctl) -> remote.get();
+    Cache NOP = (key, remote, ctl) -> remote.get().handle(
+        (content, throwable) -> {
+            final Optional<? extends Content> res;
+            if (throwable == null) {
+                res = Optional.of(content);
+            } else {
+                res = Optional.empty();
+            }
+            return res;
+        }
+    );
```

Comment:
@olegmoz hm, but is not this exactly the goal? As far as I understand we wanted to handle all the cases when content is not found for any reasons returning empty optional from the cache and 404 error for our client. 
So, maybe I should just log the exception here?

Reply:
@olenagerasimova if remote fails for whatever reason proxy should return result from cache, that's right. But to me it seems like that's responsibility of proxy, not cache. Maybe we can move error handling from cache to the proxy? Maybe we can extract `Cache` implementation with error handling code to separate class, so it could be explicitly composed with `Cache.NOP` on higher level of abstraction?